[mattisongjj] ip

mattisongjj opened the pull request:
# Bro

> "Bro." – Bro

**Bro** frees your mind from having to remember tasks. It's designed to be:
- **text-based**
- easy to learn
- FAST and *SUPER FAST* to use

### Getting Started
1. [Download Bro from here](https://github.com/mattisongjj/ip).
2. Double-click the file to run it.
3. Add your tasks.
4. Let it manage your tasks for you. 😉

### Features
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

### Sample Code
```java
public class Bro {
    public static void main(String[] args) {
        System.out.println("Hi bro");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mattisongjj committed:
Feat: Rename Duke to Bro

mattisongjj committed:
Feat: Implement echo and goodbye commands

mattisongjj committed:
Feat: Implement TaskList class for task tracking

mattisongjj committed:
Feat: Add list and add command

mattisongjj committed:
Feat: Add Task class, to encapsulate a task

mattisongjj committed:
Feat: Integrate Task class into TaskList

mattisongjj committed:
Feat: Implement marking and unmarking tasks

mattisongjj committed:
Feat: Add todo class

mattisongjj committed:
Feat: Add deadline task

mattisongjj committed:
Feat: Add event task

mattisongjj committed:
Test: Add test case for task adding

mattisongjj committed:
Chore: Add detailed error messages

mattisongjj committed:
Feat: Add support for deleting tasks

mattisongjj committed:
refactor: Add enum type for chatbot commands

mattisongjj committed:
Implement file save

mattisongjj committed:
Implement dates and times for deadline task

mattisongjj committed:
Implement Parser class

mattisongjj committed:
Implement UI class

mattisongjj committed:
Implement Storage class that handles writing and loading form file data

mattisongjj committed:
Refactor into packages

mattisongjj committed:
Merge branch 'add-gradle-support'

mattisongjj committed:
Add unit case for TaskList

mattisongjj committed:
Package app as JAR

mattisongjj committed:
Add JavaDoc comments

mattisongjj committed:
Follow Coding Standard

mattisongjj committed:
Implement Find command

yhanyi commented on README.md:
Fantastic naming sense bro!

yhanyi commented on src/main/java/bro/Bro.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    public static void main(String[] args) {
```
Maybe need add Javadoc for the main function.

yhanyi commented on src/main/java/bro/parser/Parser.java:
Maybe can add more Javadocs for this.

yhanyi commented on src/main/java/bro/parser/Parser.java:
I like the extensive use of ChatCommand enums, makes handling the commands look cleaner.

mattisongjj committed:
Add GUI with JavaFX

Soumyadip-cmd commented on src/main/java/bro/task/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+
+import java.util.ArrayList;
+
+// bro.task.TaskList class encapsulates a task list tracked by bro.Bro Bot
```
Might consider putting it as javadoc comment and adding javadoc comments for the methods

Soumyadip-cmd commented on src/main/java/bro/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+ * the content of the task and its completion status.
+ */
+public class Task {
+    private final String content;
```
Shouldn't constant names be all uppercase?

Soumyadip-cmd commented on src/main/java/bro/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+        this.isCompleted = false;
+    }
+
+    /**
+     * Returns the string representation of the task,
+     * indicating its content and completion status.
+     *
+     * @return A string representation of the task, showing
+     *         its completion status and content.
+     */
```
Would using the @inheritDoc help here?

Soumyadip-cmd commented on src/main/java/bro/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+ * load tasks from a file into a <code>TaskList</code>.
+ */
+public class Storage {
+    private final String filePath;
```
Shouldn't constant names be capitalised?

Soumyadip-cmd commented on src/main/java/bro/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+package bro.storage;
+
+import bro.BroException;
+import bro.task.*;
+
+import java.io.*;
```
Imported classes need to be imported explicitily

Soumyadip-cmd commented on src/main/java/bro/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+                }
+                Task todoTask = new TodoTask(secondArg);
+                return new CreateCommand(taskList, todoTask, storage);
+            case deadline:
```
Provided coding standards say that there should no indentation for case clauses

Soumyadip-cmd reviewed (commented):
Overall looks good. Just some minor coding standard violations

mattisongjj committed:
Add assertions to command execute responses

There are several commands, each with their own responses that gets shown on the UI once executed.
To ensure that these commands are not empty, assertions are added

mattisongjj committed:
Improve code quality

There are several variables that could be declared as final as they do not change once assigned.

Declaring them as final improves the code quality

mattisongjj committed:
Merge pull request #1 from mattisongjj/branch-A-CodeQuality

Improve code quality

mattisongjj committed:
Merge branch 'master' into branch-A-Assertions

mattisongjj committed:
Merge pull request #2 from mattisongjj/branch-A-Assertions

Add assertions

mattisongjj committed:
Implement period task

A period task is a task that has a start and end date

It can be created with the command: period <description> /from <start date> /to <end date>

mattisongjj committed:
Merge pull request #3 from mattisongjj/B-DoWithinPeriodTasks

Add period task

mattisongjj committed:
Add window title

mattisongjj committed:
Refactor parser class

mattisongjj committed:
Update readMe

mattisongjj committed:
Update readMe

mattisongjj committed:
Update README
